daemon/cmd/ipcache.go

```diff
@@ -265,17 +265,19 @@ func (d *Daemon) restoreIPCache(localPrefixes map[netip.Prefix]identity.NumericI
 			// The prefix's labels *have* been restored from the checkpoint.
 			//
 			// If the restored identity contains an exact CIDR match
-			// for this prefix, then insert its labels in to the ipcache.
+			// for this prefix, or a FQDN selector label, then insert its labels
+			// in to the ipcache.
 			//
 			// Otherwise, do nothing. The set of labels for this prefix will
-			// be recreated from other sources on startup, such as reading the FQDN
-			// checkpoint.
+			// be recreated from other sources on startup.
 			//
 			// Note that the set of labels may be more than just reserved:world
 			// and the cidr:xxx/32. For example, the CIDR may also have the apiserver
 			// or reserved:remote-node label.
 			wantLbl, _ := labels.IPStringToLabel(prefix.String())
-			if _, exists := id.Labels[wantLbl.Key]; exists {
+			_, isCIDRLabel := id.Labels[wantLbl.Key]
+			hasFQDNLabel := id.Labels.HasSource(labels.LabelSourceFQDN)
+			if isCIDRLabel || hasFQDNLabel {
```

Comment:
Honestly, let's just remove this conditional; we can always restore these prefix + labels.